include/dlaf/blas_tile.tpp

```diff
@@ -40,6 +40,55 @@ void gemm(blas::Op op_a, blas::Op op_b, T alpha, const Tile<const T, device>& a,
              c.ptr(), c.ld());
 }
 
+template <class T, Device device>
+void hemm(blas::Side side, blas::Uplo uplo, T alpha, const Tile<const T, device>& a,
+          const Tile<const T, device>& b, T beta, const Tile<T, device>& c) {
+  SizeType m;
+  SizeType n;
+  m = c.size().rows();
+  n = c.size().cols();
+
+  if (side == blas::Side::Left) {
+    DLAF_ASSERT(m == a.size().rows(), "`m` cannot be determined!", m, a);
```

Comment:
Commenting this line for all similar ones.

I'm wondering if it wouldn't be easier to use
`DLAF_ASSERT(m == a.size().rows(), m, a.size());`

My points are:
1. I don't remember what printing a tile means (maybe just shape related properties), but I think it would be easier to read if we report in the message just the values we are checking.
2. Generally, I would not add a "human" message like "`m` cannot be determined!", hoping that the condition, which is reported in the error literally, is clear enough.

Reply:
This was used previously. It should be addressed in another PR.

Reply:
This is the error printed without the "human" message:
```
[ERROR] /home/lara/Documents/DLAF/DLA-Future_trans2stdeigen_4rebase/include/dlaf/blas_tile.tpp:35 : void dlaf::tile::gemm(blas::Op, blas::Op, T, const dlaf::Tile<const T, device>&, const dlaf::Tile<const T, device>&, T, const dlaf::Tile<T, device>&) [with T = float; dlaf::Device device = (dlaf::Device)0]
m != c.size().rows()
0
size=(0, 0), ld=1
```
What do you think, @albestro? Let's ask @rasolca and @teonnik, as well.
If you agree that this is easier to read, I can modify all similar ASSERT (here or in a different PR?).

Reply:
It was already used in many place so it should be updated everywhere. It should be adressed elsewhere to avoid to add clutters to this PR.